hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientRatis.java

```diff
@@ -258,8 +258,15 @@ public XceiverClientReply watchForCommit(long index, long timeout)
           .sendWatchAsync(index, RaftProtos.ReplicationLevel.ALL_COMMITTED);
       replyFuture.get(timeout, TimeUnit.MILLISECONDS);
     } catch (Exception e) {
+      String nodes = " with Datanodes : ";
       Throwable t = HddsClientUtils.checkForException(e);
-      LOG.warn("3 way commit failed ", e);
+      for (DatanodeDetails datanodeDetails : pipeline.getNodes()) {
+        nodes += datanodeDetails.getHostName() + "["
```

Comment:
This line will thrown findbugs, lets use StringBuilder here in place of concatenation.

Reply:
done. replaced with pipeline